src/main/java/core/basesyntax/dao/FruitOperationDaoImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitOperation;
+
+public class FruitOperationDaoImpl implements FruitOperationDao {
+    @Override
+    public void add(FruitOperation fruitOperation) {
+        Storage.SHOP_STORE.add(fruitOperation);
+    }
+
+    @Override
+    public FruitOperation get(String fruit) {
+        return Storage.SHOP_STORE.stream()
+                .filter(fruitOperation -> fruitOperation.getFruit().equals(fruit))
+                .findFirst()
+                .get();
+    }
+
+    @Override
+    public void update(FruitOperation fruitOperation) {
+        FruitOperation fruitOperationFromDb = get(fruitOperation.getFruit());
+        Storage.SHOP_STORE.remove(fruitOperationFromDb);
+        add(fruitOperation);
```

Comment:
The `update` method implementation may lead to a ConcurrentModificationException if the `SHOP_STORE` is modified while it is being iterated over. Also, this implementation does not account for the case where the `get` method might throw an exception due to the element not existing. Consider a safer way to update the storage that guarantees atomicity and handles potential exceptions.